### ISSUE TITLE
Placing concatenated string into output array

### ISSUE THREAD

**Issue by blkdragn77** [bazaarvoice/jolt#698]
Input
`{
  "id": "01001",
  "name1": "Larry",
  "name2": "Bird",
  "content1": "Basketball",
  "content2": "Court",
  "content3": "Stadium"
}`

Spec
`[
  {
    "operation": "shift",
    "spec": {
      "content1": [
		    "sports[1].name",
        "player.details[0].sport"
      ],
      "content2": "sports[0].location",
      "id": "player.id",
      "name1": "player.firstname",
      "name2": "player.lastname"
    }
  },
  {
    "operation": "modify-default-beta",
    "spec": {
      "player": {
        "details": {
          "0": {
            "fullname": "=concat(@(1,name1),' ',@(1,name2))"
          }
        }
      }
    }
  },
  {
    "operation": "default",
    "spec": {
      "player": {
        "details[]": {
          "0": {
            "teamname": "Celtics"
          }
        }
      }
    }
  }
]
`

Output
`{
  "sports" : [ {
    "location" : "Court"
  }, {
    "name" : "Basketball"
  } ],
  "player" : {
    "details" : [ {
      "sport" : "Basketball",
      "teamname" : "Celtics"
    } ],
    "id" : "01001",
    "firstname" : "Larry",
    "lastname" : "Bird"
  }
}
`

I am attempting to get the full name placed in the details array in the player object.  I have a feeling it has to do with navigating the tree, but I have been unable to get the values.  If you see where this went wrong, your assistance would be greatly appreciated!

**Comment by milosimpson** [comment]
Spec
```
[
  {
    "operation": "shift",
    "spec": {
      "content1": [
		    "sports[1].name",
        "player.details[0].sport"
      ],
      "content2": "sports[0].location",
      "id": "player.id",
      "name1": "player.details[0].firstname",
      "name2": "player.details[0].lastname"
    }
  },
  {
    "operation": "modify-default-beta",
    "spec": {
      "player": {
        "details": {
          "[0]": {   // also modify deals with arrays differently than the much older "default" operation
            "fullname": "=concat(@(1,firstname),' ',@(1,lastname))"   // problem was here
          }
        }
      }
    }
  },
  {
    "operation": "default",
    "spec": {
      "player": {
        "details[]": {
          "0": {
            "teamname": "Celtics"
          }
        }
      }
    }
  }
]
```

**Comment by blkdragn77** [comment]
Ok, so you cannot use the values from the source JSON.  The values have to be placed within the array in order for them to concatenate?

**Comment by milosimpson** [comment]
The first shift moves name1 to a new place, and "name1" no longer exists.    Aka, run just the first transform step / shift, and see that "name1" does not exist anymore.

**Comment by blkdragn77** [comment]
Thank you for the information!  So, in a chain spec, the first operation can see the source JSON.  I am assuming the operations are executed in order.  So, the source JSON is available for the 1st operation, the result of that is available to the 2nd, and so on...right?  Just trying to better understand the mechanics of how operations are processed and what data is available where.

**Comment by milosimpson** [comment]
Yeah, each operation takes as input the output from the operation before it.   Its a transform pipeline.
